tests/test_server.py

```diff
@@ -0,0 +1,100 @@
+from unittest.mock import Mock, patch
+
+from flask import g
+import pytest
+
+from scripts import __version__, server
+
+# see the flask docs for an explanation of creating an example test client:
+# https://flask.palletsprojects.com/en/1.1.x/testing/#the-testing-skeleton
+
+# see also the docs on mocking the 'g' context object:
+# https://flask.palletsprojects.com/en/1.1.x/testing/#faking-resources
+
+
+@pytest.fixture
+def client():
+    # FIXME incomplete
+    server.app.config['TESTING'] = True
+
+    with server.app.test_client() as client, server.app.app_context():
+            yield client
+
+
+def test_index(client):
+    # check home page
+    rv = client.get('/')
+    assert 'version %s' % __version__ in rv.data.decode()
+    assert b'<form action="/search" method="post">' in rv.data
+    assert b'<input type="hidden" name="format" value="html"/>' in rv.data
```

Comment:
do we get unittest assertions like `assertContains()` if we wrap these in a `TestCase`, or is that Django thing? wondering if it would make assertions about the html content any cleaner...these seem OK though

Reply:
`assertContains()` is a django test convenience. I didn't see anything in the flask testing docs that would help. Agree, this isn't as clean, but workable.